### PR TITLE
Fix py3 tests

### PR DESCRIPTION
**Pull request by cpennington**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,44 +33,42 @@ help:
 	@echo '                                                                                     		'
 
 requirements.js:
-	npm install
+	npm ci
 	# Allow root for Docker
 	$(NODE_BIN)/bower install --allow-root
 
 requirements: requirements.js
-	pip install -r e2e/requirements.txt --exists-action w
```
Comment (MatthewPiatetsky) [inline Makefile:40]: why is this removed?

File: docs/test_ecommerce.rst [hunk 2]
```diff
@@ -97,17 +97,7 @@ these models, which are used for multi-tenancy.
...
-  Note: ``DISABLE_MIGRATIONS`` is broken as of June 2019.
-
-  Setting the ``DISABLE_MIGRATIONS`` variable significantly decreases the time
-  needed to run tests by creating the test database directly from Django model
-  definitions as opposed to applying the defined migrations.
-
-  .. code-block:: bash
-
-      $ DISABLE_MIGRATIONS=1 DISABLE_ACCEPTANCE_TESTS=True ./manage.py test ecommerce.courses.tests.test_utils:UtilsTests.test_get_course_info_from_catalog_cached --settings=ecommerce.settings.test --with-ignore-docstrings --logging-level=DEBUG
+      $ tox -e py35-tests -- ecommerce.courses.tests.test_utils:UtilsTests.test_get_course_info_from_catalog_cached
```
Comment (MatthewPiatetsky) [inline docs/test_ecommerce.rst:100]: why are the commands above deleted?  
does this tox command work for both python 2 and python 3 tests? or is running python 2 tests no longer supported?

File: e2e/test_auth.py [hunk 3]
```diff
@@ -1,11 +1,12 @@
 from __future__ import absolute_import
 
-import pytest
 from selenium.common.exceptions import NoSuchElementException
 
 from e2e.config import LMS_URL_ROOT, MARKETING_URL_ROOT
 from e2e.helpers import EcommerceHelpers, LmsHelpers
 
+import pytest  # isort:skip
```
Comment (MatthewPiatetsky) [inline e2e/test_auth.py:8]: is the isort:skip added everywhere because you scripted the addition of the import or something?

File: ecommerce/core/models.py [hunk 4]
```diff
@@ -534,7 +535,7 @@ def lms_user_id_with_metric(self, usage=None, allow_missing=False):
                      self.id, usage, exc_info=True)
         else:
             monitoring_utils.set_custom_metric('ecommerce_missing_lms_user_id', self.id)
-            log.warn(u'Could not find lms_user_id with metric for user %s for %s.', self.id, usage, exc_info=True)
+            log.warning(u'Could not find lms_user_id with metric for user %s for %s.', self.id, usage, exc_info=True)
```
Comment (MatthewPiatetsky) [inline ecommerce/core/models.py:538]: why did warn change to warning?

File: ecommerce/extensions/api/v2/views/vouchers.py [hunk 5]
```diff
@@ -82,7 +83,7 @@ def offers(self, request):
 
         try:
             offers_data = self.get_offers(request, voucher)
-        except (ConnectionError, SlumberBaseException, Timeout):
+        except (ReqConnectionError, SlumberBaseException, Timeout):
```
Comment (MatthewPiatetsky) [inline ecommerce/extensions/api/v2/views/vouchers.py:86]: why is ConnectionError now imported as ReqConnectionError?

File: pylintrc [hunk 6]
```diff
@@ -66,6 +66,10 @@ disable=
 # We can decide if names are invalid on our own
     invalid-name,
 
+# We use isort for import order
+    wrong-import-order,
```
Comment (MatthewPiatetsky) [inline pylintrc:71]: why are these disabled?

**Review by MatthewPiatetsky — COMMENTED** [review]
1. have you tested locally that you're still able to checkout?
2. normally I suggest people squash their commits. given that this is a lot of files, do you think all your commits stand as independent commits? i could see the value of more commits with so many files
3. i didn't really review the travis/tox/requirements changes since i'm not familiar enough, so if you feel like you need those parts reviewed i'd ask someone else
4. given the size of this PR i'm assuming some slightly higher than average risk. is there any reason that just reverting the entire PR wouldn't work if something was broken?

File: Makefile (hunk 1 above)
Comment (cpennington) [reply Makefile:40]: The `e2e.in` file was added as a dependency of `dev.in`.

File: docs/test_ecommerce.rst (hunk 2 above)
Comment (cpennington) [reply docs/test_ecommerce.rst:100]: `DISABLE_MIGRATIONS` is already baked into the tox settings.

And yes, it works as `tox -e py27-tests ...` as well, but that won't be relevant soon, since we're going to be switching entirely to python 3

File: e2e/test_auth.py (hunk 3 above)
Comment (cpennington) [reply e2e/test_auth.py:8]: No, it's because py27 and py35 isorts disagree about where pytest should go. Once we're only running python3, we can remove the `isort:skip`.

File: ecommerce/core/models.py (hunk 4 above)
Comment (cpennington) [reply ecommerce/core/models.py:538]: Because `warn` is deprecated, and `warning` isn't.

File: ecommerce/extensions/api/v2/views/vouchers.py (hunk 5 above)
Comment (cpennington) [reply ecommerce/extensions/api/v2/views/vouchers.py:86]: Because in python 3, `ConnectionError` is a builtin, and so we don't want to shadow it with the version imported from `requests`

File: pylintrc (hunk 6 above)
Comment (cpennington) [reply pylintrc:71]: Made the comment clearer. Basically, because we use isort, and isort and pylint disagree, I'm choosing to use just the isort error messages, not both.

**Comment by cpennington** [comment]
>     1. have you tested locally that you're still able to checkout?
Not yet, I've been focused on getting the unit tests updated

>     2. normally I suggest people squash their commits. given that this is a lot of files, do you think all your commits stand as independent commits? i could see the value of more commits with so many files
My intention is that all commits stand on their own. That said, there are a few towards the end that are isolated pylint or other quality fixes that could, in theory, be rolled into the commits that caused the issues.

>     3. i didn't really review the travis/tox/requirements changes since i'm not familiar enough, so if you feel like you need those parts reviewed i'd ask someone else

>     4. given the size of this PR i'm assuming some slightly higher than average risk. is there any reason that just reverting the entire PR wouldn't work if something was broken?
No, there really shouldn't be any functional changes in this PR, it should be completely revertable.

**Review by MatthewPiatetsky — APPROVED** [review]

File: .travis/docker-compose-travis.yml [hunk 7]
```diff
@@ -37,4 +36,4 @@ services:
       DB_USER: "ecomm001"
       DJANGO_SETTINGS_MODULE: "ecommerce.settings.test"
       JASMINE_HOSTNAME: "localhost"
-      JASMINE_WEB_DRIVER: "Firefox"
+      JASMINE_WEB_DRIVER: "FirefoxHeadless"
```
Comment (adzuci) [inline .travis/docker-compose-travis.yml:39]: 👍
